### DISCUSSION TITLE
few Git repos are not getting indexed

### DISCUSSION THREAD

**Discussion by anvesh-k99** [oracle/opengrok#4606] [Q&A]
Hi Team, I am trying to upgrade opengrok version to latest version and after upgrade few of the repos are not getting indexed and also I do not see xref created for those repos. I could successfully index all my repos on version 1.12.15. Can someone please help here. Below is the error that am seeing during indexing.

org.opengrok.indexer.index.IndexDatabase addIndexDatabaseForProject
SEVERE: Failed to create history cache for some repositories of project <project-name>:indexed=false,history=true: {{dir='/ngs/app/rosd/og/mnt/src/<project-name>/apple-connect-saml',type=git,historyCache=on,renamed=false,merge=true,annotation...

opengrok-indexer -J=-Xmx8g -J=-Djava.util.logging.config.file=$HOME/og/etc/logging.properties -a $HOME/og/opengrok/lib/opengrok.jar -- -c /usr/local/bin/ctags -s $HOME/og/mnt/src -d $HOME/og/data -H -P -S -G -W $HOME/og/etc/configuration.xml -U

**Comment by vladak** [comment]
Since the changes for #4456 history cache creation failures makes the indexing of related project fail. You need to investigate why the history cache failed to be generated. You can also work around that by using the `--ignoreHistoryCacheFailures` indexer option however I would not recommend that as it might have consequences to index consistency.

**Reply by anvesh-k99** [reply]
Thanks for the reply!! What we observed is empty repositories are failing for creating history cache. On opengrok version 1.12.19 it is ignoring empty repos but above that version is failing to create history cache. Is it something opengrok does not handle empty repositories.

**Reply by vladak** [reply]
What do you mean by "empty repository" exactly ?

**Reply by anvesh-k99** [reply]
I mean git repository without any files

**Reply by vladak** [reply]
I see. There are indeed couple of places in the `GitRepository` class where this will fail.

**Reply by anvesh-k99** [reply]
May I know how this can be handled, we have multiple empty git repositories and all those are not getting indexed

**Reply by vladak** [reply]
Either use the above mentioned indexer option or wait for a fix.

**Reply by anvesh-k99** [reply]
May be I will have to wait for fix as if I use the indexer option it will ignore history cache and it will not index those particular repositories. Please correct me if I am wrong.

**Reply by vladak** [reply]
If these are empty, there is nothing to index anyway.

**Reply by anvesh-k99** [reply]
Yeah but other repositories in the same org are  being skipped from indexing

**Reply by vladak** [reply]
I don't think they will be if the `--ignoreHistoryCacheFailures` is used (in related project, that is).

**Reply by vladak** [reply]
will be fixed via PR https://github.com/oracle/opengrok/pull/4608
